packages/react/time-series/src/TimeSeries.tsx

```diff
@@ -203,6 +196,32 @@ export function TimeSeries(props: TimeSeriesProps) {
         }
       }
 
+      if (chartRef.current) {
+        const datasetRef = chartRef.current.data.datasets[0]
+
+        chartRef.current.data.labels = labels
+        datasetRef.data = values
+        if (chartRef.current.options.layout?.padding) {
+          chartRef.current.options.layout.padding = styles?.canvas?.padding
+        }
+        chartRef.current.options.scales = isFormatted || isStatic ? customFormatScales : autoFormatScales
+
+        datasetRef.type = variant
+        datasetRef.backgroundColor =
+          styles?.bar?.backgroundColor || styles?.line?.backgroundColor || defaultStyles.bar.backgroundColor
+        datasetRef.borderColor = styles?.bar?.borderColor || styles?.line?.borderColor || defaultStyles.bar.borderColor
+        datasetRef.borderWidth = styles?.bar?.borderWidth || styles?.line?.borderWidth || defaultStyles.bar.borderWidth
+        datasetRef.hoverBackgroundColor =
+          styles?.bar?.hoverBackgroundColor ||
+          styles?.line?.hoverBackgroundColor ||
+          defaultStyles.bar.hoverBackgroundColor
+        datasetRef.hoverBorderColor =
+          styles?.bar?.hoverBorderColor || styles?.line?.hoverBorderColor || defaultStyles.bar.hoverBorderColor
+
+        chartRef.current.update()
+        return
```

Comment:
Make we could break up this function in multiple ones for each process like, updating, setting base options, defining options idk. Just to make this function easily readable, got it?